### PR TITLE
[AArch64] Lowering of fpmode intrinsics in DAG

### PR DESCRIPTION
**Pull request by spavloff**
LLVM intrinsics `get_fpmode`, `set_fpmode` and `reset_fpmode` operate control modes, the bits of FP environment that affect FP operations. On AArch64 these bits are in FPCR. The lowering implemented to produce code close to that of GLIBC.

### CODE REVIEW

**Comment by spavloff** [comment]
Ping.

**Comment by spavloff** [comment]
Ping.

File: llvm/lib/Target/AArch64/AArch64ISelLowering.h [hunk 1]
```diff
@@ -523,6 +523,9 @@ enum Rounding {
 // Bit position of rounding mode bits in FPCR.
 const unsigned RoundingBitsPos = 22;
 
+// Reserved bits should be preserved when modifying FPCR.
+const uint64_t ReservedFPControlBits = 0xfe0fe0f8;
```
[the remark this answers is not in the document]
Comment (spavloff) [reply llvm/lib/Target/AArch64/AArch64ISelLowering.h]: You are right, thank you. The set of reserved bits was blindly copied from glibc: https://sourceware.org/git/?p=glibc.git;a=blob;f=sysdeps/aarch64/fpu/fpu_control.h;h=263cf36c057ae91bcf1175e39f392fc4caa84c10;hb=refs/heads/master#l49.

**Comment by spavloff** [comment]
Ping.

**Comment by spavloff** [comment]
Ping.

**Comment by john-brawn-arm** [comment]
LGTM.

**Comment by spavloff** [comment]
Thanks!

**Pull request merged**